File: java/oddities.java
import java.util.Scanner;

public class oddities {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int n = scan.nextInt();
		while(n --> 0) {
			int x = scan.nextInt();
			if(Math.abs(x) % 2 == 0)
				System.out.printf("%d is even%n", x);
			else
				System.out.printf("%d is odd%n", x);
		}
		scan.close();
	}
}

File: ts/src/oddities.ts
import readline = require('readline')

let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      const num = parseInt(line)
      if (Math.abs(num) % 2 === 0) {
        console.log(`${num} is even`)
      } else {
        console.log(`${num} is odd`)
      }
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
